packages/mds-agency/utils.ts

```diff
@@ -482,7 +482,7 @@ export function computeCompositeVehicleData(payload: VehiclePayload) {
     composite.updated = event.timestamp
     composite.status = (EVENT_STATUS_MAP[event.event_type as VEHICLE_EVENT] || 'unknown') as VEHICLE_STATUS
   } else {
-    composite.status = VEHICLE_STATUSES.removed
+    composite.status = VEHICLE_STATUSES.inactive
```

Comment:
I wonder if we're ever gonna _not_ get an `event` back because we have the fall-back code now in `readPayload`. I'm preeeetty sure we'll always get an event back (even if it's the fake one). I think we can get rid of this `if (event)` block and replace it with something like:
```typescript
composite.prev_event = event.event_type
composite.updated = event.timestamp ?? null
composite.status = EVENT_STATUS_MAP[event.event_type]
```

Reply:
ok i'll stick in some logging in case we don't get an event back

Reply:
well TS seems to think we need the type guard in place

Reply:
because a `VehiclePayload` may not have an event

Reply:
I don't see how `readPayload` provides a fallback. If there is no event, `readPayload` won't make that event not null or undefined. `readPayload` only provides a fallback if the event_type is missing, not if the entire event is.

Reply:
I'd misread the logic in `readPayload`, though you were doing the compensation for a missing event there. In any case, we should still probably add `composite.prev_event = event.deregister` in this else case now.

Reply:
Ok done. I'm really not familiar with the business logic of agency--is there an equivalent of setting `prev_event` to `deregister` for the bulk vehicle reads? I only see a reference to `prev_event` in the code path for the single vehicle read.

Reply:
I don't think so -- the endpoints differ in that regard (which needs to be fixed). There's outstanding cleanup that's needed for these endpoints that I'll cut a PR for against `develop` once this PR lands.